### PR TITLE
refactor(FileSource): pre-Parse file in constructor Layer.

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
* Fetch/Parse `FileSource` data when is added.
  * Fetch in `FileSource`
  * Parse fetched data in `Layer`: inline or by callBack? 
       ➡️ **I choose inline parsing.**

## Motivation and Context
* Build `FileSource#extent` on Layer adding.
* Remove specific test for `FileSource` in ProcessingFeature.
* Avoid multi fetch/parse/convert in ProcessingFeature for `FileSource`.

### CODE REVIEW

File: src/Layer/Layer.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import * as THREE from 'three';
 import { STRATEGY_MIN_NETWORK_TRAFFIC } from 'Layer/LayerUpdateStrategy';
 import InfoLayer from 'Layer/InfoLayer';
 import Source from 'Source/Source';
+import { parseSourceData } from 'Provider/DataSourceProvider';
```
Comment (zarov) [inline src/Layer/Layer.js:5]: Do you plan on moving it in this class ?

File: test/unit/dataSourceProvider.js [hunk 2]
```diff
@@ -26,9 +25,13 @@ const holes = require('../data/geojson/holesPoints.geojson.json');
 describe('Provide in Sources', function () {
     const renderer = new Renderer();
     renderer.setClearColor();
-
-    supportedFetchers.set('image/png', () => Promise.resolve(new THREE.Texture()));
-    supportedFetchers.set('application/json', () => Promise.resolve(holes));
+    // /!\ Avoid to overload fetcher because could troubleshoot the other unit tests?
+    // formatTag to avoid it
```
Comment (zarov) [inline test/unit/dataSourceProvider.js:29]: Good idea !

File: test/unit/source.js [hunk 3]
```diff
@@ -198,17 +198,18 @@ describe('Sources', function () {
         it('should instance and use FileSource with url', function () {
             const source = new FileSource({
                 url: '..',
+                parsedData: {},
```
Comment (zarov) [inline test/unit/source.js]: If you add this, you are doing the "should instance and use FileSource with parsedData" test below.

**Review by zarov — COMMENTED** [review]
Looks good, a few things to change

File: src/Layer/Layer.js (hunk 1 above)
Comment (gchoqueux) [reply src/Layer/Layer.js:5]: I don't think so because maybe I move the responsibility to the future NodeLayer.

File: test/unit/source.js (hunk 3 above)
Comment (gchoqueux) [reply test/unit/source.js]: I will change it to fetch a reel file.

File: src/Layer/Layer.js (hunk 1 above)
Comment (zarov) [reply src/Layer/Layer.js:5]: Ok good to me

File: test/unit/source.js (hunk 3 above)
Comment (zarov) [reply test/unit/source.js]: Ok

Comment (zarov) [reply test/unit/source.js:201]: Ok

Comment (gchoqueux) [reply test/unit/source.js]: `done()`

**Review by zarov — APPROVED** [review]
Looks good, thanks !

**Pull request merged**